SimPEG/utils/model_utils.py

```diff
@@ -100,42 +100,64 @@ def ismember(a, b):
 def depth_weighting(
     mesh, reference_locs, active_cells=None, exponent=2.0, threshold=None, **kwargs
 ):
-    """A simple depth weighting function
+    r"""
+    Construct diagonal elements of a depth weighting matrix
 
-    This function is a simple form of depth weighting based off of the vertical distance
-    of mesh cell centers from the reference location(s).
-
-    This is commonly used to counteract the natural decay of potential field data at
-    depth.
+    Builds the model weights following the depth weighting strategy, a method
+    to generate weights based on the vertical distance between mesh cell
+    centers and some reference location(s).
+    Use these weights in regularizations to counteract the natural decay of
+    potential field data with depth.
 
     Parameters
     ----------
-    mesh : discretize.base.BaseMesh
-        discretize model space.
-    reference_locs : float or (n, dim) numpy.ndarray
-        the reference values for top of the points
-    active_cells : (mesh.n_cells) numpy.ndarray of bool, optional
-        index vector for the active cells on the mesh.
-        A value of ``None`` implies every cell is active.
+    mesh : :class:`discretize.base.BaseMesh`
+        Discretized model space.
+    reference_locs : float or 2d-array
+        Reference location for the depth weighting.
+        It can be a ``float``, which value is the vertical component for
+        the reference location.
+        Or it can be a 2d array, with multiple reference locations, where each
+        row should contain the coordinates of a single location point in the
+        following order: _x, _y, _z.
+        The vertical coordinate of the reference location for each cell in the
+        mesh will be obtained by the closest point in ``reference_locs`` using
+        only their horizontal coordinates.
+    active_cells : :meth:`discretize.base.BaseMesh.n_cells`, array of bools, optional
+        Index vector for the active cells on the mesh.
+        If ``None``, every cell will be assumed to be active.
+        Default ``None``.
     exponent : float, optional
-        exponent parameter for depth weighting.
-    threshold : float, optional
-        The default value is half of the smallest cell width.
+        Exponent parameter for depth weighting.
+        The exponent should match the natural decay power of the potential
+        field. For example, for gravity acceleration, set it to 2; for magnetic
+        fields, to 3.
+        Default ``2``.
+    threshold : float or None, optional
+        Threshold parameters used in the depth weighting.
+        If ``None``, it will be set to half of the smallest cell width.
+        Default ``None``.
 
     Returns
     -------
-    (n_active) numpy.ndarray
-        Normalized depth weights for the mesh, at every active cell.
+    wz : 1d-array
```

Comment:
The returned class is a `numpy.ndarray`.

so it would be:
`wz : numpy.ndarray`

Also naming the output is just fine, particularly here when you refer to the output in the notes (but in general, when the function has only a single output, it is redundant).